### DISCUSSION TITLE
Audio and/or file transfer via usb

### DISCUSSION THREAD

**Discussion by Smarnz** [bangcorrupt/hacktribe#65] [Ideas]
I understand that this is likely not easy and probably not your priority but wondered if this would be something that would ever be on the table or is this not possible. Loving hacktribe cheers man

**Comment by bangcorrupt** [comment]
I may be able to hijack the sample import/export to re-route it over SysEx.  It's not as simple as the grooves and fx as the PCM data is in separate RAM attached to the DSP.  I've started on this but it probably won't be working any time soon.  

Live audio over USB  won't happen.  It's probably not impossible, but working such a large feature into a compiled RTOS is likely not worth it.  This may be possible in a custom firmware built from scratch.

I had a crazy idea about replacing the I2S ADC/DAC with a Teensy and taking USB audio from there, but hardware mods maybe aren't for everyone.

**Comment by ecco00** [comment]
now that we're at it :D is there any chance of being able to power it via usb? (would be handy to move around with a power bank, since the unit drains batteries like crazy)

**Reply by bangcorrupt** [reply]
I've seen someone post a mod for this before, but I can't find a link now.  

I think it would work with a barrel jack on a 6 x AA (or D) battery clip, I haven't tried.

**Comment by maks** [comment]
@bangcorrupt needless to say, sample sending over Sysex would be an amazing feature! 👍🏻  
I've added having a sample editor to the TODO list for my elfer e2 companion app (well actually its now just a Hacktribe companion app! 🙂) so its ready to make use of it, if you are able to hack this into the firmware in the future.
